do_cmake.sh

```diff
@@ -72,18 +72,23 @@ elif type ccache > /dev/null 2>&1 ; then
     ARGS+=" -DWITH_CCACHE=ON"
 fi
 
-cxx_compiler="g++"
-c_compiler="gcc"
-# 20 is used for more future-proof
-for i in $(seq 20 -1 11); do
-  if type -t gcc-$i > /dev/null; then
-    cxx_compiler="g++-$i"
-    c_compiler="gcc-$i"
-    break
-  fi
-done
-ARGS+=" -DCMAKE_CXX_COMPILER=$cxx_compiler"
-ARGS+=" -DCMAKE_C_COMPILER=$c_compiler"
+if [[ "$ARGS $@" =~ "-DCMAKE_C_COMPILER" ]] && [[ "$ARGS $@" =~ "-DCMAKE_CXX_COMPILER" ]]; then
```

Comment:
no need to check for `-DCMAKE_C_COMPILER` in `$ARGS`, which is used to collect the non-user-specified args. checking both `$@` and `$ARGS` is confusing, as we are trying to appending `-DCMAKE_C_COMPILER` options to `$ARGS`.

Reply:
I'm not sure I understand why the C compiler check is redundant, can you please explain again?

Reply:
Are you suggesting that we should only check for cxx compiler and then set both cxx and c accordingly?